packages/vuetify/src/mixins/activatable/index.ts

```diff
@@ -0,0 +1,114 @@
+// Mixins
+import Delayable from '../delayable'
+import Toggleable from '../toggleable'
+
+// Utilities
+import mixins from '../../util/mixins'
+import { getSlot, getSlotType } from '../../util/helpers'
+import { consoleError } from '../../util/console'
+
+// Types
+import { PropValidator } from 'vue/types/options'
+import { VNode } from 'vue'
+
+const baseMixins = mixins(
+  Delayable,
+  Toggleable
+)
+
+/* @vue/component */
+export default baseMixins.extend({
+  name: 'activatable',
+
+  props: {
+    activator: {
+      default: null,
+      validator: (val: string | object) => {
+        return ['string', 'object'].includes(typeof val)
+      }
+    } as PropValidator<string | HTMLElement>,
+    disabled: Boolean,
+    internalActivator: Boolean,
+    openOnHover: Boolean
+  },
+
+  data: () => ({
+    activatorElement: null as null | HTMLElement,
+    activatorNode: [] as VNode[]
+  }),
+
+  watch: {
+    activator () {
+      this.activatorElement = null
+      this.getActivator()
+    }
+  },
+
+  mounted () {
+    const slotType = getSlotType(this, 'activator', true)
+
+    if (slotType && ['v-slot', 'normal'].includes(slotType)) {
+      consoleError(`The activator slot must be bound, try '<template v-slot:activator="{ on }"><v-btn v-on="on">'`, this)
+    }
+  },
+
+  methods: {
+    genActivator () {
+      const node = getSlot(this, 'activator', {
+        on: this.genActivatorListeners()
+      }) || []
+
+      this.activatorNode = node
+
+      return node
+    },
+    genActivatorListeners () {
+      if (this.disabled) return {}
+
+      const listeners: Record<string, (e: MouseEvent & KeyboardEvent) => void> = {}
+
+      if (this.openOnHover) {
+        listeners.mouseenter = (e: MouseEvent) => {
+          this.getActivator(e)
+          this.runDelay('open')
+        }
+        listeners.mouseleave = (e: MouseEvent) => {
+          this.getActivator(e)
+          this.runDelay('close')
+        }
+      } else {
+        listeners.click = (e: MouseEvent) => {
+          const activator = this.getActivator(e)
+
+          if (activator) activator.focus()
+
+          this.isActive = !this.isActive
+        }
+      }
+
+      return listeners
+    },
+    getActivator (e?: Event): HTMLElement | null {
+      // If we've already fetched the activator, re-use
+      if (this.activatorElement) return this.activatorElement
```

Comment:
`activator` and `e.target` can change

Reply:
:ok_hand:

Reply:
@johnleider can you explain how it works?

Reply:
![image](https://user-images.githubusercontent.com/9064066/56503452-17b71280-64db-11e9-9f2a-0f9695b0e831.png)

I'll look into the e.target use-case

Reply:
Works just fine:

```vue
<template>
  <div>
    <v-menu>
      <template #activator="{ on }">
        <v-btn
          v-if="value"
          v-on="on"
        >Click 1</v-btn>
        <v-btn
          v-else
          fab
          fixed
          bottom
          right
          v-on="on"
        ></v-btn>
      </template>
      <v-card>
        <v-card-text>Lorem ipsum</v-card-text>
      </v-card>
    </v-menu>

    <v-btn @click="value = !value">Toggle</v-btn>
  </div>
</template>

<script>
  export default {
    data: () => ({
      value: true
    })
  }
</script>
```